packages/oasis-actions/src/strategies/aave/adjust.ts

```diff
@@ -411,21 +422,16 @@ async function _decreaseRisk({
   /*
     Final position calculated using actual swap data and the latest market price
   */
-  // EG FROM WBTC 8 to USDC 6
-  // Convert WBTC fromWei
-  // Apply market price
-  // Convert result back to USDC at precision 6
-  const debtTokenAmountAfterSwapInWei = amountToWei(
-    amountFromWei(target.swap.fromTokenAmount, args.collateralToken.precision).div(
-      actualMarketPriceWithSlippage,
-    ),
-    args.debtToken.precision,
-  ).integerValue(BigNumber.ROUND_DOWN)
+  const debtTokenAmount = amountFromWei(
+    target.swap.fromTokenAmount,
+    args.collateralToken.precision,
+  ).div(actualMarketPriceWithSlippage)
+  const debtTokenAmountInBaseUnits = amountToWei(debtTokenAmount, args.debtToken.precision)
 
   const newDebt = {
-    amount: existingPosition.debt.amount.minus(debtTokenAmountAfterSwapInWei),
+    amount: BigNumber.max(existingPosition.debt.amount.minus(debtTokenAmountInBaseUnits), ZERO),
     precision: existingPosition.debt.precision,
-    symbol: target.position.collateral.symbol,
+    symbol: target.position.debt.symbol,
   }
```

Comment:
I had the same issues/thoughts in withdraw. What should happen when user wants to withdraw more then they have? Throw error? just return zero? 🤷

Reply:
Yea, so occasionally when decreasing we're returning more from the swap than the remaining debt balance. The swap simulation (max slippage) is always pessimistic so when the swap out performs it we have to handle this yea. Sometimes no debt left.